pkg/tsdb/query_context.go

```diff
@@ -6,9 +6,9 @@ type QueryContext struct {
 	TimeRange   *TimeRange
 	Queries     QuerySlice
 	Results     map[string]*QueryResult
-	ResultsChan chan *BatchResult
-	Lock        sync.RWMutex
-	BatchWaits  sync.WaitGroup
+	ResultsChan chan *BatchResult `json:"-"`
+	Lock        sync.RWMutex      `json:"-"`
+	BatchWaits  sync.WaitGroup    `json:"-"`
```

Comment:
is there a reason for adding these, when is QueryContext ever marshalled to json ?